cosmos/providers/dbt/core/operators.py

```diff
@@ -234,45 +223,32 @@ def run_command(self, cmd, env):
         self.exception_handling(result)
         return result
 
-    def build_and_run_cmd(self, env: dict, cmd_flags: list = None):
+    def build_and_run_cmd(
+        self, context: Context, cmd_flags: list[str] | None = None
+    ) -> SubprocessResult:
         create_default_profiles(DBT_PROFILE_PATH)
         profile, profile_vars = map_profile(
             conn_id=self.conn_id, db_override=self.db_name, schema_override=self.schema
         )
-
-        # parse dbt command
-        dbt_cmd = []
-
-        ## start with the dbt executable
-        dbt_cmd.append(self.dbt_executable_path)
-
-        ## add base cmd
+        dbt_cmd = [self.dbt_executable_path]
         if isinstance(self.base_cmd, str):
             dbt_cmd.append(self.base_cmd)
         else:
-            [dbt_cmd.append(item) for item in self.base_cmd]
-
-        # add global flags
-        for item in self.add_global_flags():
-            dbt_cmd.append(item)
-
-        ## add command specific flags
+            dbt_cmd.extend(self.base_cmd)
+        dbt_cmd.extend(self.add_global_flags())
+        # add command specific flags
         if cmd_flags:
-            for item in cmd_flags:
-                dbt_cmd.append(item)
-
-        ## add profile
-        dbt_cmd.append("--profile")
-        dbt_cmd.append(profile)
+            dbt_cmd.extend(cmd_flags)
+        # add profile
+        dbt_cmd.extend(["--profile", profile])
```

Comment:
@MikeWallis42 I'm a bit confused here. Wouldn't this go against the "DBT_PROFILES_DIR" environment variable override?